src/install.rs

```diff
@@ -52,12 +52,25 @@ pub fn install(
     if let InstanceType::AppImage = instance_type {
         fs::rename(&archive_file, &executable_path)?;
     } else {
-        archive::unpack(&archive_file, &destination)?;
+        if !cfg!(target_os = "macos") {
+            archive::unpack(&archive_file, &destination)?;
+        }    
     }
 
-    if cfg!(unix) {
+    if cfg!(target_os = "macos") {
+        let archive_path = archive_file.to_string_lossy().to_string();
```

Comment:
Whereever possible, operate on PathBuf or Path. You may turn it into a string if necessary (in that case `to_string_lossy()` will do), for example for logging.